src/data_publisher.py

```diff
@@ -9,8 +9,12 @@ class QueueManager:
     Manages a collection of queues and broadcast data to the queues.
     """
 
-    _queues: list[Queue] = []
-    _broadcast_lock: Lock = Lock()
+    _queues: list[Queue]
+    _broadcast_lock: Lock
+
+    def __init__(self) -> None:
+        self._queues = list()
+        self._broadcast_lock = Lock()
```

Comment:
I would personally do this like this

```suggestion
    def __init__(self) -> None:
        self._queues: list[Queue] = list()
        self._broadcast_lock: Lock = Lock()
```